libs/wingsdk/src/shared-aws/bucket.inflight.ts

```diff
@@ -200,15 +201,34 @@ export class BucketClient implements IBucketClient {
     }
     return list;
   }
+  /**
+   * checks if the bucket is public
+   * @returns true if the bucket is public and false otherwise
+   */
+  private async checkIfPublic(): Promise<boolean> {
+    const command = new GetPublicAccessBlockCommand({
+      Bucket: this.bucketName,
+    });
+    const resp: GetPublicAccessBlockCommandOutput = await this.s3Client.send(
+      command
+    );
+    return (
+      !resp.PublicAccessBlockConfiguration?.BlockPublicAcls &&
+      !resp.PublicAccessBlockConfiguration?.BlockPublicPolicy &&
+      !resp.PublicAccessBlockConfiguration?.RestrictPublicBuckets &&
+      !resp.PublicAccessBlockConfiguration?.IgnorePublicAcls
+    );
+  }
 
   /**
    * Returns a url to the given file.
    * @Throws if the file is not public or if object does not exist.
    */
   public async publicUrl(key: string): Promise<string> {
-    if (!this._public) {
+    if (!(await this.checkIfPublic())) {
```

Comment:
I am wondering: instead of rendering the url in line 241, can we just return the url the api sends back?

Reply:
I looked for it, and I couldn't see any direct reference to that online, only for a signed URL, do you have an idea how to do it?  
`GetPublicAccessBlockCommand` returns this type: https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-client-s3/Interface/PublicAccessBlockConfiguration/ 
which just contains booleans...

Reply:
Got it. OK, so the current implementation is reasonable.